packages/instant-apps-components/src/components/instant-apps-app-guide/instant-apps-app-guide.tsx

```diff
@@ -1,18 +1,89 @@
-import { Component, Host, h } from '@stencil/core';
+import { Component, Host, h, Prop, State } from '@stencil/core';
+import { Element, HostElement } from '@stencil/core/internal';
+import AppGuideViewModel from './AppGuide/AppGuideViewModel';
+import { AppGuidePage, AppGuideRenderType } from './AppGuide/interfaces/interfaces';
+import AppGuide_T9n from '../../assets/t9n/instant-apps-app-guide/resources.json';
+import { getMessages } from '../../utils/locale';
 
 @Component({
   tag: 'instant-apps-app-guide',
   styleUrl: 'instant-apps-app-guide.scss',
   shadow: true,
 })
 export class InstantAppsAppGuide {
+  @Element()
+  el: HTMLInstantAppsAppGuideElement;
 
-  render() {
+  @Prop()
+  withHeader: boolean;
+
+  @Prop()
+  data: AppGuidePage[];
+
+  @State() messages: typeof AppGuide_T9n;
+
+  viewModel: AppGuideViewModel = new AppGuideViewModel();
+
+  componentWillRender() {
+    this._initialize();
+  }
+
+  componentDidRender() {
+    getMessages(this);
+  }
+
+  render(): HostElement {
     return (
       <Host>
-        <slot>It Works!</slot>
+        <calcite-panel scale="s">
+          { this._renderAppGuideHeader() }
+          <calcite-carousel arrow-type={this._getArrowType()}>
+            {this._renderAppGuidePages(this.viewModel.pages)}
+          </calcite-carousel>
+        </calcite-panel>
       </Host>
-    );
+    )
   }
 
+  private _getArrowType() {
```

Comment:
can you create a type for this and add a return type to this method

Reply:
Per our conversation, I'll use the type provided by the `calcite-components` import.